packages/block-editor/src/components/media-placeholder/index.js

```diff
@@ -146,7 +145,7 @@ export class MediaPlaceholder extends Component {
 		let instructions = labels.instructions || '';
 		let title = labels.title || '';
 
-		if ( ! hasUploadPermissions && ! onSelectURL ) {
+		if ( ! mediaUpload && ! onSelectURL ) {
 			instructions = __( 'To edit this block, you need permission to upload media.' );
```

Comment:
Hi @youknowriad, 
Given what you referred:
> "role" is not the only reason the user can't upload. In a serverless gutenberg instance, upload is never allowed for instance.

Maybe this message should also be updated.

Reply:
But In this particular case, the message is right because if you're not allowed to paste URLs (there's no `onSelectURL`) the only way to add media is `mediaUpload` but if `mediaUpload` is not available, it means you don't have the rights to upload as in theory if you use the component without passing `onSelectURL` it means you at least provided a `mediaUpload` handler which means the only reason this could happen is if there's no permissions.

I get we can update with a more generic `Unable to edit this block` or something like that to simplify though.